feature[v2]: adding documentation regarding authentication in metric scalers (#260)
kedacore/keda-docs: content/docs/2.0/scalers/metrics-api.md

### Authentication Parameters

> Metrics Scaler API supported three types of authentication. API Key based authentication, basic authentication and TLS 

Reviewer: ```suggestion
Metrics Scaler API supported three types of authentication - API Key based authentication, basic authentication and TLS 
```

> authentication. You can use `TriggerAuthentication` CRD to configure the authentication by providing `authMode` along 

Reviewer: ```suggestion
authentication.

You can use `TriggerAuthentication` CRD to configure the authentication by providing `authMode` along 
```

with other necessary parameters as mentioned:

**API Key based authentication:**

> - `authMode`: It must be set to `apiKeyAuth` in case of Basic Authentication.

Reviewer: Nit - No need to use `Auth` suffix
```suggestion
- `authMode`: It must be set to `apiKey` in case of API key authentication.
```

- `apiKey`: API Key needed for authentication.
- `method`: This specifies the possible methods API Key based authentication supports. Possible values are `header` and `query`. `header` is the default method.
- `keyParamName`: This is either header key or query param used for passing apikey. Default header is `X-API-KEY` and default query param is `api_key`. 
If your implementation has different key, please specify it here.

**Basic authentication:**

> - `authMode`: It must be set to `basicAuth` in case of Basic Authentication.

Reviewer: Same nit, but optional
```suggestion
- `authMode`: It must be set to `basic` in case of basic authentication.
```

- `username`: This is a required field. Provide the username to be used for basic authentication.
- `password`: Provide the password to be used for authentication. For convenience, this has been marked optional, 
because many application implements basic auth with a username as apikey and password as empty.

**TLS authentication:**

> - `authMode`: It must be set to `tlsAuth` in case of TLS Authentication.

Reviewer: Same nit, but optional

```suggestion
- `authMode`: It must be set to `tls` in case of TLS authentication.
```

Assuming such response, Metrics API trigger will figure out that current metric value is 12.

> NOTE: The value of the metric must be json number type. The value is casted to **integer**.

> For metric scaler with API Key based authentication,

Reviewer: ```suggestion
Here is an example of a metric scaler with API Key based authentication:
```

```yaml

> 

Reviewer: ```suggestion
```

apiVersion: v1
kind: Secret
metadata:
  name: keda-metric-api-secret
  namespace: default
data:
  authMode: "apiKeyAuth"
  apiKey: "APIKEY" 
  method: "query"
  keyParamName: "QUERY_KEY"
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: keda-metric-api-creds
  namespace: default
spec:
  secretTargetRef:
    - parameter: authMode
      name: keda-metric-api-secret
      key: authMode
    - parameter: apiKey
      name: keda-metric-api-secret
      key: apiKey
    - parameter: method
      name: keda-metric-api-secret
      key: method
    - parameter: keyParamName
      name: keda-metric-api-secret
      key: keyParamName
---
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: http-scaledobject
  namespace: keda
  labels:
    deploymentName: dummy
spec:
  maxReplicaCount: 12
  scaleTargetRef:
    name: dummy
  triggers:
    - type: metrics-api
      metadata:
        targetValue: "7"
        url: "http://api:3232/components/stats"
        valueLocation: 'components.worker.tasks'
      authenticationRef:
        name: keda-metric-api-creds

> 

Reviewer: ```suggestion
```

```

For metric scaler with Basic Authentication, define the `Secret` and `TriggerAuthentication` as follows

```yaml

> 

Reviewer: ```suggestion
```

apiVersion: v1
kind: Secret
metadata:
  name: keda-metric-api-secret
  namespace: default
data:
  authMode: "basicAuth" 
  username: "username" 
  password: "password"
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: keda-metric-api-creds
  namespace: default
spec:
  secretTargetRef:
    - parameter: authMode
      name: keda-metric-api-secret
      key: authMode
    - parameter: username
      name: keda-metric-api-secret
      key: username
    - parameter: password
      name: keda-metric-api-secret
      key: password

>    

Reviewer: ```suggestion
```

```

> 

Reviewer: ```suggestion
```

For metric scaler with TLS Authentication, define the `Secret` and `TriggerAuthentication` as follows

```yaml

> 

Reviewer: ```suggestion
```

apiVersion: v1
kind: Secret
metadata:
  name: keda-metric-api-secret
  namespace: default
data:
  authMode: "tlsAuth" 
  cert: "cert" 
  key: "key"
  ca: "ca"
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: keda-metric-api-creds
  namespace: default
spec:
  secretTargetRef:
    - parameter: authMode
      name: keda-metric-api-secret
      key: authMode
    - parameter: cert
      name: keda-metric-api-secret
      key: cert
    - parameter: key
      name: keda-metric-api-secret
      key: key
    - parameter: ca
      name: keda-metric-api-secret
      key: ca

>    

Reviewer: ```suggestion
```
